### ISSUE TITLE
'composer quick-start' command fails with 'sh: composer: command not found'

### ISSUE THREAD

**Issue by kbrinner** [apigee/apigee-devportal-kickstart-drupal#225]
**Describe the bug**
Attempting to run `composer quick-start` fails with error
```
> composer install
sh: composer: command not found
Script composer install handling the quick-start event returned with error code 127
```

**To Reproduce**
1. Run `composer create-project apigee/devportal-kickstart-project:8.x-dev stock-devportal --stability dev --no-interaction`
2. Change to directory `cd stock-devportal`
3. Attempt to install using quick-start option `composer quick-start`
```
> composer install
sh: composer: command not found
Script composer install handling the quick-start event returned with error code 127
```

**Expected behavior**
Drupal installed using PHP's built in web server etc using the Drupal 8 Quick Start Command.

**Additional info**
Running `composer install` or `composer update` from within the stock-devportal directory returns 
```Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
> DrupalComposer\DrupalScaffold\Plugin::scaffold
  - .csslintrc (https://git.drupalcode.org/project/drupal/raw/8.7.5/.csslintrc): Downloading (100%)         
  - .editorconfig (https://git.drupalcode.org/project/drupal/raw/8.7.5/.editorconfig): Downloading (100%)         
  - .eslintignore (https://git.drupalcode.org/project/drupal/raw/8.7.5/.eslintignore): Downloading (100%)         
  - .eslintrc.json (https://git.drupalcode.org/project/drupal/raw/8.7.5/.eslintrc.json): Downloading (100%)         
  - .gitattributes (https://git.drupalcode.org/project/drupal/raw/8.7.5/.gitattributes): Downloading (100%)         
  - .ht.router.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/.ht.router.php): Downloading (100%)         
  - .htaccess (https://git.drupalcode.org/project/drupal/raw/8.7.5/.htaccess): Downloading (100%)         
  - index.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/index.php): Downloading (100%)         
  - robots.txt (https://git.drupalcode.org/project/drupal/raw/8.7.5/robots.txt): Downloading (100%)         
  - sites/default/default.services.yml (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/default/default.services.yml): Downloading (100%)         
  - sites/default/default.settings.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/default/default.settings.php): Downloading (100%)         
  - sites/development.services.yml (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/development.services.yml): Downloading (100%)         
  - sites/example.settings.local.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/example.settings.local.php): Downloading (100%)         
  - sites/example.sites.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/example.sites.php): Downloading (100%)         
  - update.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/update.php): Downloading (100%)         
  - web.config (https://git.drupalcode.org/project/drupal/raw/8.7.5/web.config): Downloading (100%)  
```
Attempting to run `composer quick-start` after `composer install` or `composer update` returned the same error as above 

**Version Info**
`composer show` provides the following info (only included what I thought was relevant):

- apigee/apigee_devportal_kickstart     1.3.0  
- composer/installers                   v1.6.0 
- drupal/core                           8.7.5  
- drupal/apigee_edge                    1.0.0

`composer version` returns `Composer version 1.8.5 2019-04-09 17:46:47`

**Screenshot**
<img width="1132" alt="Screen Shot 2019-07-17 at 1 52 45 PM" src="https://user-images.githubusercontent.com/957324/61410989-3a6a8480-a89a-11e9-8105-52c37930404c.png">

**Comment by jacine** [comment]
Hi @kbrinner! 

Sounds like `composer create-project...` works just fine, and `composer quick-start` fails? You shouldn't need to run `composer install` if you started with `composer create-project ...` but either way the error you are getting is odd, since composer has clearly installed some dependencies. A couple of things to try:

- I wonder if something went wrong, and not all dependencies downloaded?  You could try `composer update --lock` and see if anything additional downloads.  If so, you could try running `composer quick-start` again.
- Check to see if you have composer in your PATH. I just read somewhere that running scripts without this  could cause a problem.
- Might also be worth making sure your composer is up to date: `composer self-update`

HTH.  Let me know how it goes!

Thanks,
Jacine

**Comment by kscheirer** [comment]
I was not able to reproduce this, but try `composer -vvv quick-start` - it should give more detailed information about which step is failing.

**Comment by jacine** [comment]
I just realized looking at the screenshot you posted that somehow you don’t have version 8.x-1.3. That read me file was changed and should look like this: https://git.drupalcode.org/project/apigee_devportal_kickstart/tree/8.x-1.3 

I would try starting over in a new directory.

**Comment by kbrinner** [comment]
**TL;DR**
I tried re-installing composer per 
https://getcomposer.org/doc/00-intro.md#globally
and now it works. Sigh.

If you want more info about other things I tried before re-installing composer, feel free to read on :) 
@jacine Thank you for your responses. 

In terms of getting the correct version, I followed the instructions in this readme https://git.drupalcode.org/project/apigee_devportal_kickstart/blob/8.x-1.3/README.md, which directs you to the installation instructions here: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/get-started-with-kickstart which state:

> The following command will download Drupal 8 core, the Apigee Developer Portal Kickstart profile, and each of their dependencies, into the MY_PROJECT directory:
```
composer create-project apigee/devportal-kickstart-project:8.x-dev MY_PROJECT --stability dev --no-interaction
```
Should the `composer create-project` command be changed to use a different version? 

So, I started over with a fresh install. Here are my exact steps I took:
```
$ mkdir devportal
$ composer create-project apigee/devportal-kickstart-project:8.x-dev devportal --stability dev --no-interaction
(omitted terminal output of installation)
$ cd devportal
$ composer -vvv quick-start
Reading ./composer.json
Loading config file ./composer.json
Checked CA file /private/etc/ssl/cert.pem: valid
Executing command (/Users/kristin.brinner/projects/devportal): git branch --no-color --no-abbrev -v
Executing command (/Users/kristin.brinner/projects/devportal): git describe --exact-match --tags
Executing command (/Users/kristin.brinner/projects/devportal): git log --pretty="%H" -n1 HEAD
Executing command (/Users/kristin.brinner/projects/devportal): hg branch
Executing command (/Users/kristin.brinner/projects/devportal): fossil branch list
Executing command (/Users/kristin.brinner/projects/devportal): fossil tag list
Executing command (/Users/kristin.brinner/projects/devportal): svn info --xml
Failed to initialize global composer: Composer could not find the config file: /Users/kristin.brinner/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /Users/kristin.brinner/projects/devportal/vendor/composer/installed.json
Loading plugin Composer\Installers\Plugin
Loading plugin cweagans\Composer\Patches
Loading plugin DrupalComposer\DrupalScaffold\Plugin
Running 1.8.5 (2019-04-09 17:46:47) with PHP 7.1.23 on Darwin / 18.6.0
> quick-start: composer install
Executing command (CWD): composer install
sh: composer: command not found
Script composer install handling the quick-start event returned with error code 127
```

@kscheirer you can see the verbose error message above per your comment. 

In this new directory I just created, I also tried the following:

```
$ composer update --lock
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Generating autoload files
> DrupalComposer\DrupalScaffold\Plugin::scaffold
  - .csslintrc (https://git.drupalcode.org/project/drupal/raw/8.7.5/.csslintrc): Downloading (100%)         
  - .editorconfig (https://git.drupalcode.org/project/drupal/raw/8.7.5/.editorconfig): Downloading (100%)         
  - .eslintignore (https://git.drupalcode.org/project/drupal/raw/8.7.5/.eslintignore): Downloading (100%)         
  - .eslintrc.json (https://git.drupalcode.org/project/drupal/raw/8.7.5/.eslintrc.json): Downloading (100%)         
  - .gitattributes (https://git.drupalcode.org/project/drupal/raw/8.7.5/.gitattributes): Downloading (100%)         
  - .ht.router.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/.ht.router.php): Downloading (100%)         
  - .htaccess (https://git.drupalcode.org/project/drupal/raw/8.7.5/.htaccess): Downloading (100%)         
  - index.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/index.php): Downloading (100%)         
  - robots.txt (https://git.drupalcode.org/project/drupal/raw/8.7.5/robots.txt): Downloading (100%)         
  - sites/default/default.services.yml (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/default/default.services.yml): Downloading (100%)         
  - sites/default/default.settings.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/default/default.settings.php): Downloading (100%)         
  - sites/development.services.yml (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/development.services.yml): Downloading (100%)         
  - sites/example.settings.local.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/example.settings.local.php): Downloading (100%)         
  - sites/example.sites.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/sites/example.sites.php): Downloading (100%)         
  - update.php (https://git.drupalcode.org/project/drupal/raw/8.7.5/update.php): Downloading (100%)         
  - web.config (https://git.drupalcode.org/project/drupal/raw/8.7.5/web.config): Downloading (100%) 

```

But still got the same error when trying to run `composer quick-start`

I also did the following:

```
$ composer self-update
Updating to version 1.8.6 (stable channel).
   Downloading (100%)         
Use composer self-update --rollback to return to version 1.8.5
```

But again still got the same error when trying to run `composer quick-start`.

This part of that verbose error caught my attention as I don't understand what it is looking for here:
`Failed to initialize global composer: Composer could not find the config file: /Users/kristin.brinner/.composer/composer.json`

This message just seemed wrong, so I tried re-installing composer per 
https://getcomposer.org/doc/00-intro.md#globally
And now it is all working. I included the above info so if someone else is going through something similar maybe it will be helpful.

**Comment by jacine** [comment]
Hey @kbrinner! 

Thanks so much for letting us know.  This is great info.

I'm glad you got it working!

Maybe we should specifically recommend a global installation in prerequisites, since Composer docs are a little wishy washy about it, and don't recommend one or the other.  @jbennett13 What do you think?

Jacine 😄
